src/Codeception/Command/Clean.php

```diff
@@ -20,13 +19,34 @@ class Clean extends Command
 
     public function getDescription()
     {
-        return 'Cleans or creates _output directory';
+        return 'Recursively cleans log and generated code';
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $output->writeln("<info>Cleaning up " . Configuration::outputDir() . "...</info>");
-        FileSystem::doEmptyDir(Configuration::outputDir());
+        $projectDir = Configuration::projectDir();
+        $this->cleanProjectsRecursively($output, $projectDir);
         $output->writeln("Done");
     }
+
+    private function cleanProjectsRecursively(OutputInterface $output, $projectDir)
+    {
+        $config = Configuration::config($projectDir);
+
+        $logDir = $projectDir . DIRECTORY_SEPARATOR . $config['paths']['log'];
+        $output->writeln("<info>Cleaning up log " . $logDir . "...</info>");
+        FileSystem::doEmptyDir($logDir);
+
+        if (!empty($config['paths']['helpers'])) {
```

Comment:
Clean on all systems cleans generated code, that is literally the point of it. Users are expected to run "mvn clean build package" (java/mvn), "make clean; make build; make install" (c/make), ant is the same etc.

I think if you want to have a cut down option then it can be added as a separate Command but clean standard should return the directory/files to what it would have been if it was freshly checked out.

If you want I can add a cleannogen or something but I think clean actual should carry the same behaviour as standard/expected.

Reply:
`mvn clean build package`
I don't know how this is related to this project...

> Command but clean standard should return the directory/files to what it would have been if it was freshly checked out.

If we follow this logic all user created config files should also be cleaned up by this command. 

Sorry, code generation is a tricky question and we don't have any standards in dealing with it. But I assume that `clean` can bring more problems to developers when it will be implemented this way. I don't want `clean` to potentially break the setup.

Reply:
The mvn/make are just examples of other automated systems. I was trying to highlight that there is a user expectation of what "clean" means. In all other systems the accepted practice is to clean, build, run. You have to because it's the only way to be sure that every run on every machine is the same, because that's what you need in a testing environment. I suspect that users on big projects had to write all the glue to do a proper clean before running as I did.

If you are checking in user generated config files then they are not user edited config files any more, they are project artefacts and shouldn't be cleaned.

`don't have any standards in dealing with it` you said codeception is not perfect, so maybe it's time to start standardising. The simplest is don't put configuration files into a generated folder that may not exist!

I suppose to keep BC, I could keep leave `clean` as it was (non-recursive, not-deleting generated code) and add a `realclean` which behaves like users on big projects expect (recursive and deleting generated files). If you don't like `realclean` you can pick something else. You'll get what you want and we'll get what we need!